### PR TITLE
Dither code maintenance

### PR DESCRIPTION
**Pull request by jenshannoschwalm**
Tested performance CPU vs SSE2 on current compiler, there is now a 20% performance gain for plain cpu code over sse2, so a clear go to remove the sse2 specific code.

Code cleanup as current common dt style, some const added.

### CODE REVIEW

**Comment by victoryforce** [comment]
I have the impression that soon dt will completely get rid of SSE2 code, because with modern compilers it does not bring any benefit, to say the least. 😃

**Comment by ralfbrown** [comment]
We are definitely headed in that direction.  Some of the code will require considerable contortions to make the compiler vectorize it as effectively as the SSE code does, however.

File: src/iop/dither.c [hunk 1]
```diff
@@ -160,8 +157,12 @@ static inline float _rgb_to_gray(const float *const restrict val)
 #ifdef _OPENMP
 #pragma omp declare simd
 #endif
-static inline void nearest_color(float *const restrict val, float *const restrict err, int graymode,
-                                 const float f, const float rf)
+static inline void nearest_color(
```
Comment (ralfbrown) [inline src/iop/dither.c:160]: This function should also get a leading underscore since you're changing things to be consistent with dt's style.

Comment (jenshannoschwalm) [reply src/iop/dither.c:160]: :-)

**Review by TurboGit — APPROVED** [review]
All good and no regression on the integration testsuite.

**Comment by jenshannoschwalm** [comment]
Release notes

I would recommend to summarize **all** commits 

Various code cleanups removing sse2 specific code with improved performance

**Pull request merged**